src/node/internal/crypto_dh.ts

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+//
+// Copyright Joyent, Inc. and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+/* todo: the following is adopted code, enabling linting one day */
+/* eslint-disable */
+
+'use strict';
+
+import {
+  Buffer,
+} from 'node-internal:internal_buffer';
+
+import { default as cryptoImpl } from 'node-internal:crypto';
+type ArrayLike = cryptoImpl.ArrayLike;
+
+import {
+  ERR_CRYPTO_ECDH_INVALID_PUBLIC_KEY,
+  ERR_INVALID_ARG_TYPE,
+} from 'node-internal:internal_errors';
+
+import {
+  validateInt32,
+} from 'node-internal:validators';
+
+import {
+  isArrayBufferView,
+  isAnyArrayBuffer
+} from 'node-internal:internal_types';
+
+import {
+  getArrayBufferOrView,
+  toBuf,
+  kHandle,
+} from 'node-internal:crypto_util';
+
+const DH_GENERATOR = 2;
+
+interface DiffieHellman {
+  [kHandle]: cryptoImpl.DiffieHellmanHandle;
+}
+
+let DiffieHellman = function (this: any, sizeOrKey: number|ArrayLike, keyEncoding?: number|string,
+                              generator?: number|ArrayLike, genEncoding?: string): DiffieHellman {
```

Comment:
Why `let DiffieHellman = function ...` and not simply `function DiffieHellman(...`?

Also, I believe `this: DiffieHellman` should work on that correct?